pkg/api/util/ipc_endpoint.go

```diff
@@ -59,8 +59,11 @@ func (end *IPCEndpoint) DoGet(options ...GetOption) ([]byte, error) {
 		if errStr, found := errMap["error"]; found {
 			return nil, errors.New(errStr)
 		}
-
-		return nil, fmt.Errorf("could not reach agent: %v\nMake sure the agent is running before requesting the runtime configuration and contact support if you continue having issues", err)
+		netErr := new(net.OpError)
+		if errors.As(err, &netErr) {
```

Comment:
**💬 suggestion:** ‏I believe you can do that 

```suggestion
	    if netErr, ok := err.(*net.OpError); ok {
```

Reply:
This is not quite equivalent as `errors.As` will unwrap errors, while the typing conversion will only check the top level one

Reply:
I tried what you suggested but it's not making the desirable behaviour, it's not matching the if statement. I think it's because the errors in go are stored in a tree-like structure and in this case it's not possible to cast it. The `As` method instead is searching for the provided type in this tree structure. (ref: https://pkg.go.dev/errors#As) @pierreguilmin can you confirm what I'm saying ?

Reply:
I think you tagged the wrong Pierre haha

Reply:
But about your message yeah that's also my guess, so you should probably stick to `errors.As`

Reply:
@pgimalac (this time it's the good one) I answered before I saw you were already in the thread ^^'